### ISSUE TITLE
Solution for someone who relied solely on load_from_yahoo for price data for backtesting?

### ISSUE THREAD

**Issue by drhay53** [quantopian/zipline#1876]
Hello,

I realize things are in flux as lots of people try to figure out what they're doing without the yahoo finance API. I for one am a bit lost. I basically exclusively used load_from_yahoo to get all of my price data, and I had frozen my code to use a fairly old zipline (0.7.0) which I was confident was working exactly as I intended. This is from before the deprecation of transforms (I sent a plea for help to the mailing list yesterday).

So now, all of my trades are completely frozen until I figure out how to deal with the loss of the yahoo API for price data. It's got me fairly stressed out.

My question here is: is there an option for retrieving price data which returns the same format object as load_from_yahoo did? Without the ability to even run my code, it's hard to check that I'm doing things right with some other source (also, I don't prefer pandas, so I am not confident in reproducing the old load_from_yahoo behavior myself). 

I'm afraid that I'm looking at a daunting full re-write... but if there is a similar function to load_from_yahoo already in existence or planned, then I can hold off until the anaconda build is released for OSX.

**Comment by FreddieV4** [comment]
Hey @drhay53 I saw your post in the mailing list and now have an answer for you.

**Addressing `load_from_yahoo`:**

There currently is not an option for retrieving pricing data in the same format as `load_from_yahoo` in the zipline codebase.

However, it is possible for you to implement your own [data bundle](http://www.zipline.io/bundles.html) which can load pricing data from Yahoo. _Note that the notion of data bundles doesn't exist in `0.7.0`._ You can see what the previous Yahoo data bundle looked like (which was also removed in `1.1.1`) [here](https://github.com/quantopian/zipline/blob/1.1.0/zipline/data/bundles/yahoo.py). If you choose to make a new data bundle for Yahoo, it would be awesome if you opened up a PR for that 🙂 . You could use [pandas-datareader](https://github.com/pydata/pandas-datareader) to get Yahoo data in the form of a DataFrame, when implementing your own data bundle.

You can also use `pandas-datareader` in another way with https://github.com/quantopian/zipline/pull/1860, which is a PR to add a `csvdir` data bundle that allows you to read `csv` files (which you would get from `pandas-datareader`). Until we merge the `csvdir` bundle and cut a new release, you'll need to checkout the branch that PR is on in order to use the bundle. 

**Addressing `zipline.transforms`**

If you want to use minutely data, you would want to use [data.history()](https://www.quantopian.com/help#api-data-history) in conjunction with some numpy/pandas function like `avg()`

Otherwise, the closest thing to using transforms would be the `Pipeline API`. You can find info on that in several places:
- https://www.quantopian.com/posts/introducing-the-pipeline-api
- https://www.quantopian.com/tutorials/pipeline
- http://www.zipline.io/appendix.html#pipeline-api
- https://www.quantopian.com/help#pipeline-title

So, it seems like to me that you'll need to re-write your algorithm using the latest Zipline API. Luckily there are some resources to help you out here, like the [Beginner Tutorial](https://github.com/quantopian/zipline/blob/master/docs/notebooks/tutorial.ipynb), and the [Q2 Migration Guide](https://www.quantopian.com/quantopian2/migration)

**Comment by drhay53** [comment]
Thank you @FreddieV4, I am just using the daily adjusted close; it looks like I'm up against a full re-write with the new zipline API. Refactoring the code is probably justified anyway. 

I had some trouble getting zipline to build when I checked out master earlier today; I am guessing it will be quite a while before I am able to get anything together. My free time is greatly reduced compared to when I first wrote the algorithm. 

Thanks for the resources, and hopefully I'll be able to work through all of the issues.
